services/user-tenant-service/src/main/java/com/sourcefuse/jarc/services/usertenantservice/DTO/Role.java

```diff
@@ -0,0 +1,54 @@
+package com.sourcefuse.jarc.services.usertenantservice.DTO;
+
+import com.sourcefuse.jarc.services.usertenantservice.commons.UserModifiableEntity;
+import com.sourcefuse.jarc.services.usertenantservice.enums.RoleType;
+import jakarta.persistence.*;
+import lombok.*;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.UUID;
+
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+@Table(name = "roles", schema = "main")
+public class Role extends UserModifiableEntity implements Serializable {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.UUID)
+    private UUID id;
+
+    @Column(nullable = false)
+    private String name;
+
+
+    @Column(nullable = false, name = "role_type")
+    //doubt::
+    private RoleType roleType;
```

Comment:
![image](https://user-images.githubusercontent.com/46295805/233788894-38ae24f7-dadb-4f7f-96bf-a466bfd97e63.png)

This can lead to inconsistency in the database. Scenario:
If a user is stored in the db with roletype 'Automation', db value will be 7 (not because the enum has value 7, but instead its index value is 7 in the declared list of enums. If We at later times add another enum key in between the list, the db value of 7 will then map to some other roletype).

Solution:
This can be easily avoided:
1. By always creating new entry at the end of the enum 
OR
2. By storing the enum with its actual value passed in the constructor.

I prefer 2nd approach because the 1st one creates extra dependency on the devs.

Reply:
This should be done for all the enum types which are being stored in the database as integers.

Reply:
PFA images to help you with a more clear understanding:

A tenant is created with status 'ACTIVE' stored as 1 in the db
![image](https://user-images.githubusercontent.com/46295805/233789775-18decd66-74ac-446f-b164-8c22c05518da.png)

Let us add a new entry in the TenantStatus enum (anywhere, but at the last position)
![image](https://user-images.githubusercontent.com/46295805/233789828-b0f8be61-25f6-4adf-bfd8-043b85022605.png)

Now, when the application is restarted and GET request is made to fetch the tenants, all of them will have status set as INACTIVE
![image](https://user-images.githubusercontent.com/46295805/233789912-da8c3cbb-4d0c-4678-879f-ea6e337f169a.png)

Hope this gives you a broader understanding of the issue.

Reply:
done